apps/ecommerce/frontend/src/types.ts

```diff
@@ -98,3 +98,12 @@ export interface DialogInvocationParameters extends KeyValueMap {
   linkType: string;
   fieldType: FieldType;
 }
+
+// the variant states of ProductCard, for now, labeled by location
+export type ProductCardType = 'field' | 'dialog';
+
+export type ExternalResourceError = {
+  error: string;
+  errorMessage: string;
+  errorStatus: number;
+};
```

Comment:
conflicted about this, I find the current setup of `useExternalResource` hook, with `error`, `errorMessage`, and `errorStatus` being returned as a bit confusing because I expect error to be the `error` object that encapsulates all of this. Any thoughts here appreciated!

Reply:
What you did here I think is a great placeholder for now